File: VB/VB/DXExample.Module/FilterResourcesByOwnersController.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Data.Filtering

Namespace DXExample.Module
	Partial Public Class FilterResourcesByOwnersController
		Inherits ViewController
		Public Sub New()
			TargetObjectType = GetType(UserResource)
			TargetViewType = ViewType.ListView
		End Sub
		Protected Overrides Sub OnActivated()
			MyBase.OnActivated()
			Dim adminRole As Role = View.ObjectSpace.Session.FindObject(Of Role)(New BinaryOperator("Name", "Administrators"))
			If (CType(SecuritySystem.CurrentUser, CustomUser)).Roles.Lookup(adminRole.Oid) Is Nothing Then
				CType(View, ListView).CollectionSource.Criteria("ByUsers") = CriteriaOperator.Parse("Owners[Oid = ?] OR Owners.Count = 0", SecuritySystem.CurrentUserId)
			End If
		End Sub
	End Class
End Namespace

File: CS/DXExample.Module/FilterResourcesByOwnersController.cs

using DevExpress.ExpressApp;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Data.Filtering;

namespace DXExample.Module {
    public partial class FilterResourcesByOwnersController : ViewController {
        public FilterResourcesByOwnersController() {
            TargetObjectType = typeof(UserResource);
            TargetViewType = ViewType.ListView;
        }
        protected override void OnActivated() {
            base.OnActivated();
            Role adminRole = View.ObjectSpace.FindObject<Role>(new BinaryOperator("Name", "Administrators"));
            if (((CustomUser)SecuritySystem.CurrentUser).Roles.Lookup(adminRole.Oid) == null) {
                ((ListView)View).CollectionSource.Criteria["ByUsers"] = CriteriaOperator.Parse("Owners[Oid = ?] OR Owners.Count = 0", SecuritySystem.CurrentUserId);
            }
        }
    }
}
